orchagent/bfdorch.cpp

```diff
@@ -410,12 +430,24 @@ bool BfdOrch::create_bfd_session(const string& key, const vector<FieldValueTuple
         }
 
         attrs.emplace_back(attr);
+        attr_idx++;
     }
 
     fvVector.emplace_back("state", session_state_lookup.at(SAI_BFD_SESSION_STATE_DOWN));
 
     sai_object_id_t bfd_session_id = SAI_NULL_OBJECT_ID;
-    sai_status_t status = sai_bfd_api->create_bfd_session(&bfd_session_id, gSwitchId, (uint32_t)attrs.size(), attrs.data());
+    sai_status_t status = sai_bfd_api->create_bfd_session(&bfd_session_id, gSwitchId, attr_idx, attrs.data());
+
+    for (int retry = 0; (status != SAI_STATUS_SUCCESS) && (retry < NUM_BFD_SRCPORT_RETRIES); retry++)
```

Comment:
This is a corner case scenario. I think we should handle it in a separate function and invoke if status is Failure.  You can loop the attributes to find the src port attr index.  Managing index by attr_id++ is also error prone when someone adds a new attribute.

Reply:
Updated